src/main/java/htsjdk/samtools/SamFileValidator.java

```diff
@@ -566,11 +570,36 @@ private void validateHeader(final SAMFileHeader fileHeader) {
         }
     }
 
+    /**
+     * Number of warnings during SAM file validation
+     *
+     * @return number of warnings
+     */
+    public int getNumWarnings() {
+        return this.numWarnings;
+    }
+
+    /**
+     * Number of errors during SAM file validation
+     *
+     * @return number of errors
+     */
+    public int getNumErrors() {
+        return this.numErrors;
+    }
+
     private void addError(final SAMValidationError error) {
         // Just ignore an error if it's of a type we're not interested in
         if (this.errorsToIgnore.contains(error.getType())) return;
 
-        if (this.ignoreWarnings && error.getType().severity == SAMValidationError.Severity.WARNING) return;
+        if ( error.getType().severity == SAMValidationError.Severity.WARNING) {
+            if ( this.ignoreWarnings ) {
+                return;
+            }
+            this.numWarnings++;
+        } else {
```

Comment:
Changed to a `switch` statement that handles each known error type and throws a `SAMException` if it's an unknown error.